### PR TITLE
CT-3240 Added reopening-complaint feature

### PR DESCRIPTION
**Pull request by ymao2**
## Description

This PR is to allow Branston user to reopen a closed complaint case and reset a new deadline
## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots

Screenshot 2021-01-27 at 09 08 20

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3240

### CODE REVIEW

File: app/views/cases/offender_sar_complaint/reopen.html.slim [hunk 1]
```diff
@@ -0,0 +1,15 @@
+- content_for :heading, flush: true
+  = t('cases.offender_sar_complaint.reopen_reset_external_deadline_heading')
+
+= render partial: 'layouts/header'
+
+= form_for @case, url: confirm_reopen_case_sar_offender_complaint_path(@case), as: :offender_sar_complaint do |f|
+
+  .form-group
+    = f.gov_uk_date_field :external_deadline, { \
+        legend_text: '',
```
Comment (digitalali-moj) [inline app/views/cases/offender_sar_complaint/reopen.html.slim:10]: Not sure how this renders, can you please double check this meets accessibility or let me know how to get there? Thanks.

Comment (ymao2) [reply app/views/cases/offender_sar_complaint/reopen.html.slim:10]: I didn't pay attention to this,  thank you for spotting this!   I think maybe I just add one? 
The steps for get this page is 
1 - create an complaint case
2 - proceed the case to close, 
3 - once the case is closed, you should see 'Reopen' button, click it you will see the page

Comment (ymao2) [reply app/views/cases/offender_sar_complaint/reopen.html.slim:10]: I added a screenshot for this PR, @digitalali-moj , that is how the deadline looks like

Comment (ymao2) [reply app/views/cases/offender_sar_complaint/reopen.html.slim:10]: I added the legend_text, the new looks like below,  let me know whether it is ok, thank you!
Screenshot 2021-01-28 at 10 59 32

Comment (digitalali-moj) [reply app/views/cases/offender_sar_complaint/reopen.html.slim:10]: Looks great to me! thanks!

**Review by willmcb — APPROVED** [review]
Great work LGTM 👍

**Pull request merged**